### ISSUE TITLE
Custom upd

### ISSUE THREAD

**Issue by amirburbea** [michaelwittig/node-q#31]
My company's KDB team has a non standard 'upd' call that will push more than two values.

You have the following code in your node-q library:
`
if (err === undefined && Array.isArray(o) && o[0] === "upd") { self.emit('upd', o[1], o[2]); }
`

Can this be changed to:
`
if (err === undefined && Array.isArray(o) && o[0] === "upd") { events.EventEmitter.prototype.emit.apply(self, o); }
`

or simply `self.emit(...o);` since pretty much every version of node today supports that syntax.

Basically allowing the array to be a variable length??

**Comment by michaelwittig** [comment]
`self.emit(...o)` would be nice but I try to be node 0.10 backward compatible. This would require node 6.x?

**Comment by michaelwittig** [comment]
released as `node-q@2.3.0`

**Comment by amirburbea** [comment]
Thanks, that's super helpful.  Appreciate the quick turnaround.
